### DISCUSSION TITLE
Getting error while dealing with large dataset

### DISCUSSION THREAD

**Discussion by saleharashid01** [AammarTufail/python_ka_chilla2023#103] [Q&A] [answered]
![getting error in model training](https://user-images.githubusercontent.com/121869148/217193024-6d3c5207-98a8-4125-9532-2b3cb858df1a.jpg)
![train](https://user-images.githubusercontent.com/121869148/217194721-fe097d1d-a5ca-43ba-8208-9e6b66ca69ad.jpg)

**Comment by Muhammad-Ali-Butt** [comment, marked as answer]
From the looks of it , seems like you're trying to get the best model based upon recall score?
can you explain what exactly you're trying to do ?

**Reply by Muhammad-Ali-Butt** [reply]
also the error you are encountering is due to presence of continuous variables which the model cannot deal with so you have to convert the continuous to numerical either by label encoding , onehot encoding and @AammarTufail taught us about it in machine learning videos you can check them out to refresh the concept or find another way based upon your data

**Reply by saleharashid01** [reply]
ok great, it worked (Jazak Allah) with label encoding,
(I was trying best score, parameter(grid search) 


###  Scikit learn label encoding
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
le.fit(dataset['New_deaths'])
dataset['New_deaths'] =le.transform(dataset['New_deaths'])

**Reply by saleharashid01** [reply]
@Muhammad-Ali-Butt 

when i run Best Score model, it worked for linear in my case and give  value in negative, but i move towards KNN, Decision tree, randomforest,,.. it shows me warnings and say that best score (nan, nan, nan...as shown in fig)... what does it mean,?
![error](https://user-images.githubusercontent.com/121869148/217226220-ab855034-5483-4280-8e4f-b7ab753111aa.jpg)

**Reply by Muhammad-Ali-Butt** [reply]
you have nan values in your data try checking them by `df.isnull().sum() * 100 / len(df)` and then filling them or completely dropping the column if it has more than 70% nan values. the other error is that you are using binary classification for multiclass classification hence you should change average='macro' or 'weighted' and then try again.

**Reply by Muhammad-Ali-Butt** [reply]
> ok great, it worked (Jazak Allah) with label encoding, (I was trying best score, parameter(grid search)

wa iyyakum glad it helped :blush:

**Comment by nouman073** [comment]
apply the label encoding on y

**Reply by nouman073** [reply]
same issue i faced i applied this methed then my issue solved 

from sklearn import preprocessing
from sklearn import utils

label_encoder = preprocessing.LabelEncoder()
y_train = label_encoder.fit_transform(y_train

**Comment by AzamBukhari** [comment]
![Screenshot_2023-02-11-03-03-48-908_com linkedin android](https://user-images.githubusercontent.com/121688289/218235885-8554fb7c-4081-4ef9-9626-3ad70afbd256.jpg)
